src/pages/StatePartner Dashboard/Dashboard/index.js

```diff
@@ -0,0 +1,253 @@
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+import { useSelector } from "react-redux";
+import { METHODS } from "../../../services/api";
+import { Link } from "react-router-dom";
+import { PATHS } from "../../../constant";
+import { hasOneFrom } from "../../../common/utils";
+import { Redirect } from "react-router-dom";
+import "../style.scss";
+
+function Dashboard({ stateId }) {
+  const user = useSelector(({ User }) => User);
+  const [districtPartner, setDistrictPartner] = useState([]);
+  const [districtPartnerData, setDistrictPartnerData] = useState([]);
+  const [filterData, setFilterData] = useState([]);
+  const [value, setValue] = useState();
+  const [isRegion, setIsRegion] = useState();
+  const [region, setRegionData] = useState();
+  const [regionDistrictPartner, setRegionDistrictPartner] = useState([]);
+  const [district, setDistrict] = useState();
+
+  useEffect(() => {
+    axios({
+      method: METHODS.GET,
+      url: `${process.env.REACT_APP_MERAKI_URL}/partners/${stateId}/groups`,
+      headers: {
+        accept: "application/json",
+        Authorization: user.data.token,
+      },
+    }).then((res) => {
+      setDistrictPartner(res.data);
+      setIsRegion(
+        res.data.partner_groups_data[0].partner_group_name.includes("_REGION")
+      );
+      setDistrictPartnerData(res.data.partner_groups_data);
+      setRegionData(res.data.partner_groups_data);
+
+      res.data.partner_groups_data[0].partner_group_name.includes("_REGION") &&
+        setRegionDistrictPartner(
+          res.data.partner_groups_data[0].partner_groups_data
+            .partner_groups_data
+        );
+
+      if (
+        res.data.partner_groups_data[0].partner_group_name.includes("_REGION")
+      ) {
+        setFilterData([
+          res.data.partner_groups_data[0].partner_groups_data
+            .partner_groups_data[0],
+        ]);
+      } else {
+        setFilterData([res.data.partner_groups_data[0]]);
+      }
+    });
+  }, [stateId]);
+
+  const handleSelcet = (e) => {
+    const stateFilterData = districtPartnerData.filter(
+      (el) => el.partner_group_name === e.target.value
+    );
+    setFilterData(stateFilterData);
+    setValue(e.target.value);
+  };
+
+  const selectRegion = (e) => {
+    let regionPartner;
+    region.filter((item) => {
+      if (item.partner_group_name === e.target.value) {
+        regionPartner = item.partner_groups_data.partner_groups_data;
+      }
+    });
```

Comment:
I wouldn't use `filter` here as the purpose of that method is to create a new subarray from the existing one.  Additionally, assuming there is only one item with the `partner_group_name` target value, you can stop as soon as you find it rather than continuing through the Array as `filter` would do.  In this case, you could use the following: 
```suggestion
    const regionItem = region.find((item) => item.partner_group_name === e.target.value); 
    const regionPartner = regionItem && regionItem.partner_groups_data.partner_groups_data;
```